### DISCUSSION TITLE
How to set default storage policy for database?

### DISCUSSION THREAD

**Discussion by OkenKhuman** [ClickHouse/ClickHouse#68096] [Q&A]
Is there a way to set default storage policy for database?

I know I can set the storage policy for each table on creation but I would like to set it on database level so that each table would that particular policy by default.

The storage policy I am trying to create deals with s3 disk.

**Comment by kemoid** [comment]
try in the config.xml or any xml file in the path /etc/clickhouse-server/
   <default_storage_policy>encrypted_policy</default_storage_policy>

**Comment by tr0k** [comment]
I think the original question was about how to set the default storage policy at the database level, not cluster-wide.

It looks like the setting can be defined at the DB level, but it doesn't seem to take effect, at least in version 25.8.8.26, when creating a table without explicitly specifying storage_policy:
```
CREATE DATABASE s3db
ON CLUSTER replicated
ENGINE = Atomic
SETTINGS
    default_table_engine = 'MergeTree',
    storage_policy = 'tiered_s3';
```
```
SELECT
    name,
    engine,
    default_table_engine,
    storage_policy
FROM system.databases

4. │ s3db               │ Atomic │ /var/lib/clickhouse/store/ │ store/2e8/XXX-95e6-4abb-8642-XXX/ │ XXX-95e6-4abb-8642-XXX │ Atomic SETTINGS default_table_engine = 'MergeTree', storage_policy = 'tiered_s3' │
```
